README.md

```diff
@@ -0,0 +1,494 @@
+
+If log lines exist in the current buffer, it is pushed onto a send queue, and a new buffer is created.  The send queue is processed and uploaded to LogDNA.
+
+If no work needs to be done, the `cleared` event is immediately emitted.
+
+
+### `logger.log(statement[, options])`
+
+* `statement` [`<String>`][] | [`<Object>`][] - Text or object of the log entry.  Objects are serialized.
+* `options` [`<String>`][] | [`<Object>`][] - A string representing a [level](#supported-log-levels), or object:
+    * `level` [`<String>`][] - Desired [level](#supported-log-levels) for the current message. **Default:** `logger.level`
+    * `app` [`<String>`][] - App name to use for the current message. **Default:** `logger.app`
+    * `env` [`<String>`][] - Environement name to use for the current message. **Default:** `logger.env`
+    * `timestamp` [`<Number>`][] - Epoch ms time to use for the current message. Must be within 24 hours. **Default:** `Date.now()`
+    * `context` [`<Object>`][] - Synonym for `meta`, but mutually exclusive; Ignored if `meta` exists.
+    * `indexMeta` [`<Boolean>`][] - Allows for the `meta` to be searchable in LogDNA. **Default:** `logger.indexMeta`
+    * `meta` [`<Object>`][] - Per-message meta data.  Combined with key-values created with `addMetaProperty`
+* Emits: [warn](#event-warn), [error](#event-error)
+
+Sends a string or object to LogDNA for storage.  If the [convenience methods](#convenience-methods) are used, they call this function
+under the hood, so the options are the same.  The only difference is that `level` is automatically set in the convenience methods.
+
+
+### `logger.removeMetaProperty(key)`
+
+* `key` [`<String>`][] - The key (and implied value) to be remove from the global `meta` object.
+* Emits: [`warn`](#event-warn)
+
+Attempts to remove the given key from the global `meta` data object.  If the key is not found, `warn` is emitted.
+
+
+## How Log Lines are Sent to LogDNA
+
+In default operation, when `log` functions are called, the line is added to a buffer to cut down on HTTP traffic to the server.  
+The buffer is flushed every `flushIntervalMs` milliseconds, or if the log line lengths grow beyond `flushLimit`.
+
+When `flush` fires (or is called manually), the current buffer is put onto a send queue, and a new buffer is started.  The send queue begins
+sending to LogDNA.  It will continue to send without pausing or honoring `flushIntervalMs` as long as there are buffers in the send queue.  
+When the send queue is empty, `cleared` is emitted.
+
+### Handling Errors
+
+* `timeout` or `500` errors will be retried using an [exponential backoff strategy](#exponential-backoff-strategy), but will
+  also emit `error` events along the way.
+* User-level errors (such as `400`) will not be retried because they most likely would never be successful (if the message is deemed invalid),
+  and `error` events are emitted for these also.
```

Comment:
```suggestion
  and `error` events are emitted for these events, also.
```

Reply:
hrm, `events` isn't really the right word.  It's an error that emits an event.  If you want to specify the noun, perhaps `are emitted for these errors, also.`